site/src/pages/CreateTemplatePage/UploadTemplateView.tsx

```diff
@@ -51,7 +53,14 @@ export const UploadTemplateView: FC<CreateTemplatePageViewProps> = ({
 			jobError={isJobError ? error.job.error : undefined}
 			logs={templateVersionLogsQuery.data}
 			upload={{
-				onUpload: uploadFileMutation.mutateAsync,
+				onUpload: async (file: File) => {
+					try {
+						await uploadFileMutation.mutateAsync(file);
+					} catch (error) {
```

Comment:
I'm not familiar with this query library, but what I found is that even though you can set `onError`, an exception returned from `mutateAsync` isn't caught unless you explicitly catch it. So, If I displayed the error as part of `onError` in `useMutation`, I'd still need to wrap `mutateAsync` in a try block, with an empty catch, which seems pointless.

Reply:
You did it correctly. We usually use `mutateAsync` to keep the error handling close to where the mutation is called.